scripts/lint-metrics.sh

```
@@ -0,0 +1,160 @@
+if [ -n "${GITHUB_STEP_SUMMARY:-}" ]; then
+  {
+    printf '## rust-code-analysis: all checks pass\n\n'
+    printf '| Metric | Threshold |\n'
+    printf '|--------|----------|\n'
+    printf '| Cyclomatic Complexity | <=%s |\n'  "$CC_MAX"
+    printf '| Cognitive Complexity  | <=%s |\n'  "$COGNITIVE_MAX"
+    printf '| Function Arguments    | <=%s |\n'  "$NARGS_MAX"
+    printf '| Exit Points           | <=%s |\n'  "$NEXITS_MAX"
+    printf '| Maintainability Index | >=%s |\n'  "$MI_MIN"
+    printf '| Source Lines of Code  | <=%s |\n'  "$SLOC_MAX"
+    printf '\nAll functions in `src/` are within policy thresholds.\n'
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)